sql/rpl_binlog_sender.cc

```diff
@@ -665,6 +665,19 @@ int Binlog_sender::wait_new_events(my_off_t log_pos) {
   int ret = 0;
   PSI_stage_info old_stage;
 
+  // case: sleep before locking and waiting for new data
+  if (unlikely(opt_slave_dump_thread_wait_sleep_usec != 0)) {
```

Comment:
The original code had

```
dump_thread_wait_sleep_usec= get_dump_thread_wait_sleep(thd);
```

But this is reading directly from the opt_slave_dump_thread_wait_sleep_usec, which seems wrong because that's the setting based on the master's configuration, and not the setting based on the slave's configuration.

It seems like the master's configuration setting would override any setting requested by the slave. If the master's setting was 10s, then even if the slave did not want any sleeps, the master would still be sleeping 10s.

Reply:
Sorry, I did not understand that this command should be issued from a slave to the master.